java/integTest/src/test/java/glide/BaseClientTests.java

```diff
@@ -0,0 +1,416 @@
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import glide.api.BaseClient;
+import glide.api.RedisClient;
+import glide.api.RedisClusterClient;
+import glide.api.models.commands.SetOptions;
+import glide.api.models.configuration.NodeAddress;
+import glide.api.models.configuration.RedisClientConfiguration;
+import glide.api.models.configuration.RedisClusterClientConfiguration;
+import glide.api.models.exceptions.RequestException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import lombok.SneakyThrows;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class BaseClientTests {
+
+    private static final String KEY_NAME = "key";
+    private static final String INITIAL_VALUE = "VALUE";
+    private static final String ANOTHER_VALUE = "VALUE2";
+
+    private static RedisClient standaloneClient = null;
+    private static RedisClusterClient clusterClient = null;
+
+    private static List<BaseClient> clients = new ArrayList<>();
+
+    @BeforeAll
+    @SneakyThrows
+    public static void init() {
+        standaloneClient =
+                RedisClient.CreateClient(
+                                RedisClientConfiguration.builder()
+                                        .address(
+                                                NodeAddress.builder().port(TestConfiguration.STANDALONE_PORTS[0]).build())
+                                        .build())
+                        .get(10, SECONDS);
+
+        clusterClient =
+                RedisClusterClient.CreateClient(
+                                RedisClusterClientConfiguration.builder()
+                                        .address(NodeAddress.builder().port(TestConfiguration.CLUSTER_PORTS[0]).build())
+                                        .requestTimeout(5000)
+                                        .build())
+                        .get(10, SECONDS);
+
+        clients.add(standaloneClient);
+        clients.add(clusterClient);
+    }
+
+    @AfterAll
+    @SneakyThrows
+    public static void teardown() {
+        standaloneClient.close();
+        clusterClient.close();
+    }
+
+    @Test
+    @SneakyThrows
+    public void ping() {
+        for (BaseClient client : clients) {
+            var data = client.ping().get(10, SECONDS);
+            assertEquals("PONG", data);
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void ping_with_message() {
+        for (BaseClient client : clients) {
```

Comment:
you should consider making this a @ParameterizedTest
and pass in the pair of clients as arguments. 

Like: 
```
private List<Arguments> provideClients() {
        return List.of(
            Arguments.of(standaloneClient),
            Arguments.of(clusterClient)
        );
    }

    @ParameterizedTest
    @MethodSource("provideClients")
    public void test(BaseClient client) {
            var data = client.ping("H3LL0").get(10, SECONDS);
            assertEquals("H3LL0", data);
     }
```

Reply:
Will check it out.